climpred/smoothing.py

```diff
@@ -0,0 +1,157 @@
+                res = xr.merge(da_list)
+                return res
+
+            res = _variable_dataarray_to_dataset(res)
+        return res
+
+
+except:
+    print('xesmf couldnt be loaded.')
+
+
+def spatial_smoothing_xrcoarsen(ds, coarsen_dict=None, how='mean'):
+    """Apply spatial smoothing by regridding to `boxsize` grid.
+
+    Reference:
+      * Goddard, L., A. Kumar, A. Solomon, D. Smith, G. Boer, P.
+            Gonzalez, V. Kharin, et al. “A Verification Framework for
+            Interannual-to-Decadal Predictions Experiments.” Climate
+            Dynamics 40, no. 1–2 (January 1, 2013): 245–72.
+            https://doi.org/10/f4jjvf.
+
+    Args:
+        ds (xr.object): input. xr.DataArray prefered.
+        coarsen_dict (dict): coarsen spatial latitudes.
+        how (str): aggregation type for coarsening. default: 'mean'
+
+    Returns:
+        ds_smoothed (xr.object): boxsize-regridded input
+
+    """
+    if coarsen_dict is None:
+        # guess spatial dims
+        spatial_dims_to_smooth = ds.dims.drop(['time', 'lead', 'member', 'init'])
+        # write coarsen to dict to coarsen similar to 5x5 degree
+        pass  # not implemented
+    # check whether coarsen dims are possible
+    for dim in coarsen_dict:
+        if dim not in ds.dims:
+            raise ValueError(dim, 'not in ds')
+        else:
+            if ds[dim].size % coarsen_dict[dim] != 0:
+                raise ValueError(
+                    coarsen_dict[dim], 'does not divide', ds[dim].size, 'in', dim
+                )
+    ds_out = ds.coarsen(coarsen_dict)
+    if how == 'mean':
+        ds_out = ds_out.mean()
+    elif how == 'sum':
+        ds_out = ds_out.sum()
+    return ds_out
+
+
+def temporal_smoothing(ds, smooth=4, dim='time', how='mean', rename_dim=True):
+    """Apply temporal smoothing by creating rolling smooth-timestep means.
+
+    Reference:
+      * Goddard, L., A. Kumar, A. Solomon, D. Smith, G. Boer, P.
+            Gonzalez, V. Kharin, et al. “A Verification Framework for
+            Interannual-to-Decadal Predictions Experiments.” Climate
+            Dynamics 40, no. 1–2 (January 1, 2013): 245–72.
+            https://doi.org/10/f4jjvf.
+
+    Args:
+        ds (xr.object): input.
+        smooth (int): length of smoothing of timesteps.
+                      Defaults to 4 (see Goddard et al. 2013).
+        dim (str): temporal dimension to be smoothed. default: 'time'
+        how (str): aggregation type for coarsening. default: 'mean'
+
+    Returns:
+        ds_smoothed (xr.object): input with `smooth` timesteps less and
+                                 labeling '1-(smooth-1)','...', ... .
+
+    """
+    ds_smoothed = ds.rolling({dim: smooth}, center=False)
+    if how == 'mean':
+        ds_smoothed = ds_smoothed.mean()
+    elif how == 'sum':
+        ds_smoothed = ds_smoothed.sum()
+    # remove first all-nans
+    ds_smoothed = ds_smoothed.isel({dim: slice(smooth - 1, None)})
+    if rename_dim:
```

Comment:
might be useful not to do if origin labels are needed to calculate prediction skill. tobetested.